src/base/utils/net.cpp

```diff
@@ -38,5 +40,42 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isValidParsedSubnet(const Subnet &subnet)
+        {
+            const Subnet invalid = qMakePair(QHostAddress(), -1);
+            return subnet != invalid;
```

Comment:
I don't understand your perseverance. Why this function? Why would you create a couple and compare the parameter with it, if it is known that only the first items are compared?
Maybe I'm picky... I don't see any sense in this function. @qbittorrent/demigods, judge us, please.

Reply:
[QHostAddress::parseSubnet](https://code.woboq.org/qt5/qtbase/src/network/kernel/qhostaddress.cpp.html#1120) returns that pair when invalid parameters are passed to it. Therefore I am checking directly against the expected "invalid" value.

Reply:
Thoughts? I'd like to move this PR forward.

Reply:
The one of valid solutions is implementing the following wrapper:
```c++
bool parseSubnet(const QString &subnetStr, Subnet &subnet)
{
    const Subnet invalid = qMakePair(QHostAddress(), -1);
    subnet = QHostAddress::parseSubnet(subnetStr);
    return subnet != invalid;
}
```